src/main/java/com/edmunds/tools/databricks/maven/BaseDatabricksJobMojo.java

```diff
@@ -71,264 +71,160 @@
-            Template temp = getFreemarkerConfiguration(templateLoader).getTemplate(dbJobFile.getName());
-            temp.process(jobTemplateModel, stringWriter);
-        } catch (IOException | TemplateException e) {
-            throw new MojoExecutionException(String.format("Failed to process job file as template: [%s]\nFreemarker message:\n%s", dbJobFile.getAbsolutePath(), e.getMessage()), e);
-        }
-
-        return stringWriter.toString();
-    }
-
-    protected JobTemplateModel getJobTemplateModel() throws MojoExecutionException {
-        if (StringUtils.isBlank(databricksRepo)) {
-            throw new MojoExecutionException("databricksRepo property is missing");
-        }
-        return new JobTemplateModel(project, environment, databricksRepo, databricksRepoKey, prefixToStrip);
-    }
-
-    String getJobSettingsFromTemplate(String templateText, JobTemplateModel jobTemplateModel) throws MojoExecutionException {
-        StringWriter stringWriter = new StringWriter();
-        try {
-            StringTemplateLoader templateLoader = new StringTemplateLoader();
-            templateLoader.putTemplate("defaultTemplate", templateText);
-
-            Template temp = getFreemarkerConfiguration(templateLoader).getTemplate("defaultTemplate");
-            temp.process(jobTemplateModel, stringWriter);
-        } catch (IOException | TemplateException e) {
-            throw new MojoExecutionException(String.format("Failed to process job template: [%s]\nFreemarker message:\n%s", templateText, e.getMessage()), e);
-        }
-
-        return stringWriter.toString();
-    }
-
-    Configuration getFreemarkerConfiguration(TemplateLoader templateLoader) throws IOException {
-        Configuration cfg = new Configuration(Configuration.VERSION_2_3_22);
-        cfg.setTemplateLoader(templateLoader);
-        cfg.setDefaultEncoding("UTF-8");
-        cfg.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
-
-        return cfg;
-    }
-
-
-    JobSettingsDTO[] buildJobSettingsDTOsWithDefault() throws MojoExecutionException {
-        JobTemplateModel jobTemplateModel = getJobTemplateModel();
-        String jobSettings = getJobSettingsFromTemplate(jobTemplateModel);
-        if (jobSettings == null) {
-            return new JobSettingsDTO[]{};
-        }
-
-        JobSettingsDTO defaultJobSettingDTO = defaultJobSettingDTO();
-
-        JobSettingsDTO[] jobSettingsDTOS = deserializeJobSettingsDTOs(jobSettings);
-        for (JobSettingsDTO settingsDTO : jobSettingsDTOS) {
-            try {
-                fillInDefaultJobSettings(settingsDTO, defaultJobSettingDTO, jobTemplateModel);
-            } catch (JsonProcessingException e) {
-                throw new MojoExecutionException(String.format("Fail to fill empty-value with default"), e);
+    protected TemplateModelSupplier<JobTemplateModel> createTemplateModelSupplier() {
```

Comment:
I believe this JobTemplateModel is defined in a different class too, just checking?

Reply:
I've introduced a kind of Supplier<T> because we cannot directly instantiate a generic type inside SettingsUtils class due to type erasure (if I understand your question correctly), now we're feeding SettingsUtils with that Supplier instead